### PR TITLE
Synthesizing assignment

### PR DESCRIPTION
**Pull request by leoc1114**

### CODE REVIEW

**Comment by cheuschober** [comment]
@leoc1114, You can run your tests locally too! Try to do that before sending it here. If it runs locally on your machine it will run here too.

**Comment by leoc1114** [comment]
Its funny, I ran the tests, with pylint and sh.runtests.sh, it came out ok. Except with Jenkins.


Sent from Samsung Mobile


-------- Original message --------

**Comment by cheuschober** [comment]
@leoc1114 ... Hang out, then. Perhaps the butler is confused. Let me tinker with him. I'll update you if anything's up

**Comment by cheuschober** [comment]
First week of the semester always seems to have a bug or two

**Comment by leoc1114** [comment]
I see. Thank you.


Sent from Samsung Mobile


-------- Original message --------

**Comment by cheuschober** [comment]
retest this please

**Comment by leoc1114** [comment]
I guess I didn't correct test_task_03.py, as stated by Jenkins. I'm doing that now, and it says unable to import task_03.


Sent from Samsung Mobile


-------- Original message --------

**Comment by cheuschober** [comment]
@leoc1114 This most recent run correct. I'm not sure what's happening, exactly. But you're not the first to see it. Looks like jenkins is being impatient when it first wakes up and might be timing out. If you ever get an error message that just makes zero sense, you can make a comment that says `retest this please` and it'll trigger another jenkins test.

**Comment by cheuschober** [comment]
@leoc1114 Actually... looking at your commits more closely, jenkins didn't go crazy. There's a difference between `print VARIABLENAME` and `VARIABLENAME =`

**Comment by cheuschober** [comment]
These last two runs look fine. task_06 is waiting to be done.

**Comment by leoc1114** [comment]
I passed, thank you. But not sure if I sent it correctly, to GitHub.


Sent from Samsung Mobile


-------- Original message --------

**Comment by cheuschober** [comment]
@leoc1114, from what I can see it looks good.

**Comment by leoc1114** [comment]
Thank you so much : )


Sent from Samsung Mobile


-------- Original message --------

File: task_01.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    1. Python can be used for almost any task you can think of.
+    2. It has a huge library of available modules.
+    3. Pip extends that library even more.
+    4. Why aren't you using it already! Go write some Python!
+
+
+.. table:: Sources
+
+    =========    ===============
+    Name               Contact
```
Comment (cheuschober) [inline task_01.rst]: The alignment in your table is still off, along with your heading underlines, along with your nested bullet start-point. Whitespace and alignment is important in python and restructured text. You must make sure they align consistently.

File: task_01.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    me@mymachine $ python my_file_name.py
+
+The Syntax
+=============
+Python syntax is very easy to write. Take a look at this simple statement:
+
+.. code :: python
+
+    >>>print 'Hello!'
```
Comment (cheuschober) [inline task_01.rst]: You shoudl have a space between `>>>` and `print`

File: task_01.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+Django
+==========
+
+One of the world's leading content management systems for the web. To learn more
+about Django, visit:
+
+https://www.djangoproject.com/
+
+Pyramid
+===========
```
Comment (cheuschober) [inline task_01.rst]: Also note that in the document, not all headers are at the same level.

File: tests/task_03.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (cheuschober) [inline tests/task_03.py]: This file was unnecessary. I hope it was just a mistaken creation.

File: task_01.rst [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+Pyramid
+^^^^^^^^^^
+
+A relative newcomer to the web framework. Pyramid began as part of the Pylons 
+project and has an interesting (though challenging)
+
+Other Details
+===============
+
+    1. Python can be used for almost any task you can think of.
```
Comment (cheuschober) [inline task_01.rst]: Good but you don't have to indent bullets, they can align with your headers / paragraphs. The only time they need to be indented is when they're nested in something else.

File: task_01.rst [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+Python has a ton of excellent tools available for use.
+
+Docutils
+-------------
+
+A fantastic documentation generator that interprets reStructuredText and turns it 
+into rich documents.
+
+Web Frameworks
+-----------------------------
```
Comment (cheuschober) [inline task_01.rst]: All of your header under/overlines should be the same length as the actual text in the header. This one, for example, is much too long.

File: task_01.rst [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    
+    - Always read the manual before taking questions
+    
+    - Be succint in your questions
+
+      - Don't give unnecessary details if they aren't needed
+      
+      - Don't start a question with an apology or an announcement that you're going to
+
+        ask the question. Just ask the question!
```
Comment (cheuschober) [inline task_01.rst]: Missing bold here!

**Pull request merged**